ark/utils/data_utils.py

```diff
@@ -55,60 +52,82 @@ def save_fov_images(fovs, data_dir, img_xr, sub_dir=None, name_suffix=''):
     if not os.path.exists(save_dir):
         os.makedirs(save_dir)
 
-    # retrieve the image for each fov
-    for fov in fovs:
-        fov_img_data = img_xr.loc[fov, ...].values
+    # define the file name as the fov name with the name suffix appended
+    fov_file = fov + name_suffix + '.tiff'
+
+    # save the image to data_dir
+    io.imsave(os.path.join(save_dir, fov_file), img_data, check_contrast=False)
+
+
+def relabel_segmentation(labeled_image, labels_dict):
+    """Takes a labeled image and translates its labels according to a dictionary.
+
+    Returns the relabeled array (according to the dictionary).
+
+    Args:
+        labeled_image (numpy.ndarray):
+            2D numpy array of labeled cell objects.
+        labels_dict (dict):
+            a mapping between labeled cells and their clusters.
+    Returns:
+        numpy.ndarray:
+            The image mapped to the designated new labels.
+    """
 
-        # define the file name as the fov name with the name suffix appended
-        fov_file = fov + name_suffix + '.tiff'
+    img = np.copy(labeled_image)
+    unique_cell_ids = np.unique(labeled_image)
+    unique_cell_ids = unique_cell_ids[np.nonzero(unique_cell_ids)]
 
-        # save the image to data_dir
-        io.imsave(os.path.join(save_dir, fov_file), fov_img_data, check_contrast=False)
+    default_label = max(labels_dict.values()) + 1
+    for cell_id in unique_cell_ids:
```

Comment:
Isn't this doing the same thing as your optimization PR? Can this be combined with that newly optimized function?

Reply:
@ngreenwald yeah I didn't get the chance to merge the changes in yet.